### PR TITLE
DROOLS-1706 Allow marshalling of DMN Extension Elements

### PR DESCRIPTION
**Pull request by tarilabs**
Please note it is necessary for the marshaller to know (at runtime) the composition of the namespaces, which is stored in the in-memory object Definitions, in order to assign the namespaces and prefixes correctly.

### CODE REVIEW

**Review by etirelli — APPROVED** [review]

File: kie-dmn/kie-dmn-api/src/main/java/org/kie/dmn/api/marshalling/v1_1/DMNExtensionRegister.java [hunk 1]
```diff
@@ -17,10 +17,14 @@
...
 
 import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.io.xml.QNameMap;
 
 
 public interface DMNExtensionRegister {
 
     public void registerExtensionConverters(XStream xstream);
 
+    default void beforeMarshal(Object o, QNameMap qmap) {
```
Comment (tarilabs) [inline kie-dmn/kie-dmn-api/src/main/java/org/kie/dmn/api/marshalling/v1_1/DMNExtensionRegister.java:27]: One last thing, I realized if we don't like this method here too much, we could have a 2 level interface as usual (public vs an internal one) and I move this method there.
Doesn't change however the logic, for which I need to have access to the marshalled object just before marshalling in order to setup correctly at runtime the QNameMap

Comment (etirelli) [reply kie-dmn/kie-dmn-api/src/main/java/org/kie/dmn/api/marshalling/v1_1/DMNExtensionRegister.java:27]: ok, looks fine

**Review by etirelli — APPROVED** [review]

**Review by baldimir — APPROVED** [review]
+1

**Pull request merged**